geoh5py/groups/property_group.py

```diff
@@ -75,32 +73,34 @@ class PropertyGroup:
     def __init__(  # pylint: disable=too-many-arguments
         self,
         parent: ObjectBase,
-        name: str | None = None,
+        association: str | DataAssociationEnum | None = None,
+        allow_delete: bool = True,
+        name: str = "Property Group",
         on_file: bool = False,
         uid: UUID | None = None,
-        property_group_type: GroupTypeEnum = GroupTypeEnum.UNKNOWN,
-        **kwargs,
+        property_group_type: GroupTypeEnum | str = GroupTypeEnum.UNKNOWN,
+        properties: list[UUID | Data] | None = None,
+        **_,
     ):
-        self._on_file: bool = False
-        self._allow_delete = True
-        self._association: DataAssociationEnum | None = None
-        self._properties: list[UUID] | None = None
+        self._parent: ObjectBase = self.validate_parent(parent)
+        self._property_group_type = self.validate_group_type(property_group_type)
 
+        self.allow_delete = allow_delete
         self.name = name or "property_group"
+        self.on_file: bool = on_file
         self.uid = uid or uuid4()
-        self.property_group_type = property_group_type
 
-        # define the parent
-        if not hasattr(parent, "_property_groups"):
-            raise AttributeError(
-                f"Parent {parent} must have a 'property_groups' attribute"
-            )
-        self._parent: ObjectBase = parent
-        parent.add_children([self])
+        if isinstance(properties, Iterable):
+            properties_list = [self.reference_to_data(prop) for prop in properties]
+        else:
+            properties_list = None
```

Comment:
so basicaly, users can pass anuthing to properties without raising an error.

Why not changing the previsous properties setter to a private self._properties = _validate_properties function?

Reply:
Because `validate_properties` calls your `validate_data` method, and checks for the association to match. Here I just want to get a formatted list, so that validate association can set the association if not provided.